ClientLib/src/main/java/com/o3dr/android/client/utils/data/tlog/TLogParser.java

```diff
@@ -0,0 +1,349 @@
+                File file = new File(uri.toString());
+                DataInputStream in = null;
+                LinkedList<Event> eventList = new LinkedList<>();
+                try {
+                    in = new DataInputStream(new BufferedInputStream(new FileInputStream(file)));
+                    while (in.available() > 0) {
+                        eventList.add(next(in));
+                    }
+                    callback.onResult(eventList);
+                } catch (EOFException e) {
+                    callback.onResult(eventList);
+                } catch (FileNotFoundException e) {
+                    callback.onFailed(e);
+                } catch (IOException e) {
+                    callback.onFailed(e);
+                } finally {
+                    if (in != null) {
+                        try {
+                            in.close();
+                        } catch (IOException e) {
+                            Log.e(LOG_TAG, "Failed to close file " + uri, e);
+                        }
+                    }
+                }
+            }
+        });
+    }
+
+    /**
+     * Returns a list of all events in specified TLog uri using the specified filter
+     *
+     * @param uri
+     * @param filter
+     * @param callback
+     */
+    public static void getAllEventsAsync(final URI uri, final TLogParserFilter filter, final TLogParserCallback callback) {
+        parseTlogExecutor = Executors.newSingleThreadExecutor();
+        parseTlogExecutor.execute(new Runnable() {
+            @Override
+            public void run() {
+                File file = new File(uri.toString());
+                DataInputStream in = null;
+                LinkedList<Event> eventList = new LinkedList<>();
+                try {
+                    in = new DataInputStream(new BufferedInputStream(new FileInputStream(file)));
+                    while (in.available() > 0 && filter.continueIterating()) {
+                        Event event = next(in);
+                        if (filter.addEventToList(event)) {
+                            eventList.add(event);
+                        }
+                    }
+                    sendResult(callback, eventList);
+                } catch (EOFException e) {
+                    sendResult(callback, eventList);
+                } catch (FileNotFoundException e) {
+                    sendFailed(callback, e);
+                } catch (IOException e) {
+                    sendFailed(callback, e);
+                } finally {
+                    if (in != null) {
+                        try {
+                            in.close();
+                        } catch (IOException e) {
+                            Log.e(LOG_TAG, "Failed to close file " + uri, e);
+                        }
+                    }
+                }
+            }
+        });
+    }
+
+    private static void sendResult(final TLogParserCallback callback, final List<Event> events) {
```

Comment:
@chaviw this and the method below should require a `Handler` parameter they use to dispatch the callbacks. 
Additionally, this can replace the methods from the `TLogIterator` class.